standards/research_utils.py

```diff
@@ -0,0 +1,346 @@
+        Keys : names of columns.
+        Values : The values of visualize params.
+    Returns
+    -------
+        : dict
+        key : value of name column
+        value : resulted dataframe
+    """
+    all_names = {}
+    if draw_dict is not None:
+        for key, value in draw_dict.items():
+            dataframe = dataframe[dataframe[key].isin(list([value]))]
+            if dataframe.empty:
+                raise ValueError("Incorrect column name {} or value {}.".format(key, value))
+    for names, name_df in dataframe.groupby('name'):
+        new_df = pd.DataFrame()
+        for param_names, values in name_df.groupby(parameters):
+            values['parameters'] = str(param_names).replace("'", "")
+            new_df = new_df.append(values)
+        all_names[names] = new_df
+    return all_names
+
+def _load_research(research):
+    if not isinstance(research, str):
+        return research.load_results()
+    return Research().load(research).load_results(as_dataframe=True, use_alias=False)
+
+def _get_parameters(research):
+    if isinstance(research, str):
+        with open(os.path.join(research, 'description', 'alias.json')) as res:
+            alias = json.load(res)
+        return list(alias.keys())
+
+    param_set = {}
+    for params_list in research.grid_config.alias():
+        for params in params_list:
+            param_set.update(params)
+    return list(param_set)
+
+def _prepare_results(research, hue=None, cross_val=False, aggr=False, iter_start=0, draw_dict=None, names=None):# pylint: disable=too-many-arguments
+    if cross_val or cross_val in [0, 1]:
+        results = []
+        if isinstance(research, str):
+            research = list(map(lambda res_name: os.path.join(research, res_name), os.listdir(research)))
+        parameters = _get_parameters(research[0])
+        for i, res in enumerate(research):
+            loaded_res = _load_research(res)
+            loaded_res[hue] = 0 if aggr else i
+            loaded_res = loaded_res[loaded_res['iteration'] >= iter_start]
+            results.append(loaded_res)
+        results = pd.concat(results)
+    else:
+        results = _load_research(research)
+        parameters = _get_parameters(research)
+    all_names = split_df_by_name(results, parameters, draw_dict)
+    correct_names = list(all_names.keys())
+    if len(set(correct_names) & set(names)) == 0:
+        raise ValueError('One or more names in {} are incorrect,'\
+                          ' use the names in the following list {}.'.format(names, correct_names))
+    grouped = [all_names.get(name) for name in names]
+    return grouped
+
+def draw_history(research, names, types_var, cross_val=None, aggr=False,
+                 iter_start=0, draw_dict=None): # pylint: disable=too-many-locals,too-many-arguments
+    """Draw plot with history of changes of function named `names` with values from
+    column 'types_var'. If cross validation is used, parameter `hue` allows to change
```

Comment:
There is no parameter `hue` in this function

Reply:
I'll delete this part of documentation.